src/containers/AdminCampaignEdit/sections/CampaignCannedResponsesForm/index.tsx

```diff
@@ -124,17 +124,87 @@ class CampaignCannedResponsesForm extends React.Component<InnerProps, State> {
     }
   };
 
+  cannedResponseTitleSort = (
+    { title: titleFirst }: CannedResponse,
+    { title: titleSecond }: CannedResponse
+  ) => {
+    if (titleFirst > titleSecond) {
+      return 1;
+    }
+    if (titleFirst < titleSecond) {
+      return -1;
+    }
+    return 0;
+  };
+
+  cannedResponseArraysEqual = (
+    cannedResponses1: CannedResponse[],
+    cannedResponses2: CannedResponse[]
+  ) =>
+    cannedResponses1.length === cannedResponses2.length &&
+    cannedResponses1.every(
+      (cr, idx) => cr.title === cannedResponses2[idx].title
+    );
+
+  mapToTitleDisplayOrder = (cannedResponses: CannedResponse[]) => {
+    return cannedResponses.map((cannedResponse, idx) => {
+      return { ...cannedResponse, displayOrder: idx };
+    });
+  };
+
   handleOnSaveResponse = (response: CannedResponse) => {
     const newId = Math.random()
       .toString(36)
       .replace(/[^a-zA-Z1-9]+/g, "");
 
     const { cannedResponses } = this.pendingCannedResponses();
-    const cannedResponsesToAdd = this.state.cannedResponsesToAdd.concat({
+    const cannedResponsesAlphabetical = cannedResponses.sort(
+      this.cannedResponseTitleSort
+    );
+
+    let newCannedResponse = {
```

Comment:
nit: I think this could be a `const` and then at line 189, instead of  doing: 
`
newCannedResponse = newCannedResponseWithOrder;
` 
just do: 
```   
const cannedResponsesToAdd = this.state.cannedResponsesToAdd.concat(
      newCannedResponseWithOrder
);
```
at line 204 - unless I'm missing something

Reply:
The reason I have it this way currently is because it's either:
1. We don't need to alphabetize (because a user already shifted canned responses around) so we just default to the `cannedResponses.length` for `displayOrder`
2. We do need to alphabetize and so we need to get `newCannedResponseWithOrder` and push this version with the adjusted `displayOrder`

`newCannedResponseWithOrder` is out of scope at line 204 so I think using a `let newCannedResponse` is the least bad approach i can currently think of 😅

Reply:
oh, yeah, I see now - that seems fine then!